extensions/wikia/UserLogin/js/UserLoginAjaxForm.js

```diff
@@ -1,134 +1,104 @@
-		this.cacheDOM();
-		this.wikiaForm = new WikiaForm(this.form);
-		this.inputs = this.wikiaForm.inputs;
-		this.retrieveLoginToken();
-		this.bindEvents();
-
-		if (!this.options.skipFocus) {
-			this.inputs.username.focus();
-		}
+		UserBaseAjaxForm.call(this, el, options);
 	};
 
 	/**
-	 * Cache DOM selectors
+	 * Pull in the base class's functionality
+	 * @type {UserBaseAjaxForm.prototype}
 	 */
-	UserLoginAjaxForm.prototype.cacheDOM = function () {
-		this.form = this.el.find('form');
-		this.forgotPasswordLink = this.form.find('.forgot-password');
-		this.submitButton = this.el.find('input[type=submit]');
-	};
+	UserLoginAjaxForm.prototype = Object.create(UserBaseAjaxForm.prototype);
 
 	/**
-	 * Bind form events
+	 * Set up module functionality
 	 */
-	UserLoginAjaxForm.prototype.bindEvents = function () {
-		this.form.on('submit', this.submitLogin.bind(this));
-		this.forgotPasswordLink.on('click', this.mailPassword.bind(this));
+	UserLoginAjaxForm.prototype.init = function () {
+		UserBaseAjaxForm.prototype.init.call(this);
+		this.retrieveLoginToken();
 	};
 
 	/**
-	 * Handler for login form submit
-	 * @param {Object} e jQuery event object
+	 * Callback after ajax login
+	 * @param {Object} json Response from server after ajax login
 	 */
-	UserLoginAjaxForm.prototype.submitLogin = function (e) {
-		$(window).trigger('UserLoginSubmit');
+	UserLoginAjaxForm.prototype.submitLoginHandler = function (json) {
+		var result = json.result;
+
+		UserBaseAjaxForm.prototype.submitLoginHandler.call(this, json);
 
-		this.submitButton.attr('disabled', 'disabled');
-		if (this.options.ajaxLogin) {
-			e.preventDefault();
-			this.ajaxLogin();
+		if (result === 'resetpass') {
+			this.onResetPasswordResponse();
+		} else if (result === 'unconfirm') {
+			this.onUnconfirmedEmailResponse();
+		} else if (result === 'closurerequested') {
+			this.onAccountClosureReqestResponse();
+		} else {
+			this.onErrorResponse();
 		}
 	};
 
 	/**
-	 * Make the call to the back end to log the user in via ajax
+	 * Called when a user has requested a password change
+	 * @param {Object} json Response from server
 	 */
-	UserLoginAjaxForm.prototype.ajaxLogin = function () {
-		$.nirvana.postJson(
-			'UserLoginSpecial',
-			'login',
-			{
-				loginToken: this.loginToken,
+	UserLoginAjaxForm.prototype.onResetPasswordResponse = function (json) {
+		var callback = this.options.resetpasscallback || '';
+		if (callback && typeof callback === 'function') {
+			// call with current context
+			callback.bind(this, json)();
+		} else {
+			// default implementation
+			$.post(wgScriptPath + '/wikia.php', {
+				controller: 'UserLoginSpecial',
+				method: 'changePassword',
+				format: 'html',
 				username: this.inputs.username.val(),
 				password: this.inputs.password.val(),
-				keeploggedin: this.inputs.keeploggedin.is(':checked')
-			},
-			this.submitLoginHandler.bind(this)
-		);
+				returnto: this.inputs.returnto.val(),
+				fakeGet: 1
```

Comment:
I know it's not your change, but I'm wondering why do we need this ```fakeGet``` parameter here?
I see some strange condition on PHP but to be honest I don't what would change if we remove it completely.